entity-framework/core/what-is-new/ef-core-5.0/whatsnew.md

```diff
@@ -2,17 +2,545 @@
+
+EF Core 5.0 introduces both .NET events and an EF Core interceptor triggered when SaveChanges is called.
+
+The events are simple to use; for example:
+
+```c#
+context.SavingChanges += (sender, args) =>
+{
+    Console.WriteLine($"Saving changes for {((DbContext)sender).Database.GetConnectionString()}");
+};
+
+context.SavedChanges += (sender, args) =>
+{
+    Console.WriteLine($"Saved {args.EntitiesSavedCount} changes for {((DbContext)sender).Database.GetConnectionString()}");
+};
+```
+
+Notice that:
+* The event sender is the `DbContext` instance
+* The args for the `SavedChanges` event contains the number of entities saved to the database
+
+The interceptor is defined by `ISaveChangesInterceptor`, but it is often convienient to inherit from `SaveChangesInterceptor` to avoid implementing every method. For example:
+
+```c#
+public class MySaveChangesInterceptor : SaveChangesInterceptor
+{
+    public override InterceptionResult<int> SavingChanges(
+        DbContextEventData eventData,
+        InterceptionResult<int> result)
+    {
+        Console.WriteLine($"Saving changes for {eventData.Context.Database.GetConnectionString()}");
+
+        return result;
+    }
+
+    public override ValueTask<InterceptionResult<int>> SavingChangesAsync(
+        DbContextEventData eventData,
+        InterceptionResult<int> result,
+        CancellationToken cancellationToken = new CancellationToken())
+    {
+        Console.WriteLine($"Saving changes asynchronously for {eventData.Context.Database.GetConnectionString()}");
+
+        return new ValueTask<InterceptionResult<int>>(result);
+    }
+}
+```
+
+Notice that:
+* The interceptor has both sync and async methods. This can be useful if you need to perform async I/O, such as writing to an audit server.
+* The interceptor allows SaveChanges to be skipped using the `InterceptionResult` mechanism common to all interceptors.
+
+The downside of interceptors is that they must be registered on the DbContext when it is being constructed. For example:
+
+```c#
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        => optionsBuilder
+            .AddInterceptors(new MySaveChangesInterceptor())
+            .UseSqlite("Data Source = test.db");
+```
+
+In contrast, the events can be registered on the DbContext instance at any time.
+
+### Exclude tables from migrations
+
+It is sometimes useful to have a single entity type mapped in multiple DbContexts. This is especially true when uising [bounded contexts](https://www.martinfowler.com/bliki/BoundedContext.html), for which it is common to have a different DbContext type for each bounded context.
+
+For example, a `User` type may be needed by both an authorization context and a reporting context. If a change is made to the `User` type, then migrations for both DbContexts will attempt to update the database. To prevent this, the model for one Context can be configured to exclude the table from its migrations.
```

Comment:
```suggestion
For example, a `User` type may be needed by both an authorization context and a reporting context. If a change is made to the `User` type, then migrations for both DbContexts will attempt to update the database. To prevent this, the model for one of the contexts can be configured to exclude the table from its migrations.
```